src/Components/AddressForm.tsx

```diff
@@ -69,7 +70,15 @@ export const AddressForm: React.FC<AddressFormProps> = ({
   touched,
   tabIndex,
 }) => {
-  const [address, setAddress] = React.useState({ ...emptyAddress, ...value })
```

Comment:
The address form in shipping uses the [`value`](https://github.com/artsy/force/blob/c3b5a8df47e2dac2ece9590c2d26af5b9d669125/src/Apps/Order/Routes/Shipping/index.tsx#L848) prop to render input values. The address verification flow [updates the address](https://github.com/artsy/force/blob/c3b5a8df47e2dac2ece9590c2d26af5b9d669125/src/Apps/Order/Routes/Shipping/index.tsx#L841) after accepting a suggested address, and it triggers re-rendering of the address form with the updated `value` prop. However, the initial state in `useState` here only gets used in the initial render and not in re-render. `address` becomes out-of-sync with `value` and remains the original address, although accepting the suggested address passes an updated `value` prop.

This follows the doc to [update states when props change](https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes).

Reply:
I wonder if there is a better way to not control form data from both internal state and props passed in. 🤔

Reply:
Looks great! I wonder if there is an edge case here where an updated address does not pass our form's validations. It would be good to have an idea of what cases this is possible in. The main one I am aware of is places that don't use postal codes (I think the UAE is one). This probably wouldn't affect us for US release but something I'd like to understand better.

Reply:
Good callout! We should make sure the basic form validation does not conflict with Smarty address verification.